python/ray/serve/_private/http_proxy.py

```diff
@@ -75,9 +78,21 @@ async def _send_request_to_handle(handle, scope, receive, send) -> str:
             assignment_task.cancel()
         try:
             object_ref = await assignment_task
-            result = await object_ref
-            client_disconnection_task.cancel()
-            break
+            _, timed_out = await asyncio.wait(
+                [object_ref], timeout=REQUEST_PROCESSING_TIMEOUT_S
+            )
+            if timed_out:
+                logger.debug(
+                    f"Request didn't finish within {REQUEST_PROCESSING_TIMEOUT_S} "
+                    "seconds. Retrying with another replica. You can modify "
+                    'this timeout by setting the "REQUEST_PROCESSING_TIMEOUT_S" '
+                    "env var."
+                )
+                retries += 1
```

Comment:
no backoff? the other else clause has `backoff_time_s *= 1.5`

Reply:
also missing `client_disconnection_task.cancel()`

Reply:
Thanks for the catches, I added the `backoff_time_s` and the `client_disconnection_task.cancel()`.